### ISSUE TITLE
Replace componentWillMount() and componentWillReceiveProps()

### ISSUE THREAD

**Issue by etoler1841** [trendmicro-frontend/react-sidenav#34]
The following warnings are displayed in the console when using a SideNav component:

```
Warning: componentWillMount has been renamed, and is not recommended for use. See
https://fb.me/react-async-component-lifecycle-hooks for details.

* Move code with side effects to componentDidMount, and set initial state in the
constructor.
* Rename componentWillMount to UNSAFE_componentWillMount to suppress
this warning in non-strict mode. In React 17.x, only the UNSAFE_ name will work.
To rename all deprecated lifecycles to their new names, you can run `npx
react-codemod rename-unsafe-lifecycles` in your project source folder.

Please update the following components: Uncontrolled(SideNav)
```
```
Warning: componentWillReceiveProps has been renamed, and is not
recommended for use. See https://fb.me/react-async-component-lifecycle-hooks
for details.

* Move data fetching code or side effects to componentDidUpdate.
* If you're updating state whenever props change, refactor your code to use
memoization techniques or move it to static getDerivedStateFromProps. Learn
more at: https://fb.me/react-derived-state
* Rename componentWillReceiveProps to UNSAFE_componentWillReceiveProps
to suppress this warning in non-strict mode. In React 17.x, only the UNSAFE_ name
will work. To rename all deprecated lifecycles to their new names, you can run `npx
react-codemod rename-unsafe-lifecycles` in your project source folder.

Please update the following components: Uncontrolled(SideNav)
```

**Comment by ultralabed** [comment]
Isn't this resolved already?

**Comment by bnlambert** [comment]
Any update on this?

**Comment by Modulo216** [comment]
Is this repo dead?

**Comment by congdv** [comment]
Do you have any ideas to resolve this issue? I just think we just need update the uncontrollable dependency in package.json? I'm so new with js.

**Comment by cheton** [comment]
Resolved in [0.5.0](https://github.com/trendmicro-frontend/react-sidenav/releases/tag/v0.5.0)
